src/backend/distributed/transaction/distributed_transaction_management.c

```diff
@@ -0,0 +1,483 @@
+		{
+			DistributedTransactionBackendData *activeBackend =
+				(DistributedTransactionBackendData *) palloc(
+					sizeof(DistributedTransactionBackendData));
+
+			SpinLockAcquire(&currentBackend->mutex);
+
+			activeBackend->databaseId = currentBackend->databaseId;
+			activeBackend->transactionId.initiatorNodeIdentifier =
+				currentBackend->transactionId.initiatorNodeIdentifier;
+			activeBackend->transactionId.transactionId =
+				currentBackend->transactionId.transactionId;
+			activeBackend->transactionId.timestamp =
+				currentBackend->transactionId.timestamp;
+
+			SpinLockRelease(&currentBackend->mutex);
+
+			activeTransactionList = lappend(activeTransactionList, activeBackend);
+		}
+	}
+
+	return activeTransactionList;
+}
+
+
+/*
+ * GenerateDistributedTransactionIdTuple returns a datum of pseudo-generated heaptuple
+ * for the distributed transaction id provided by the parameters.
+ */
+Datum
+GenerateDistributedTransactionIdTuple(Oid databaseId, uint64 initiatorNodeIdentifier,
+									  uint64 transactionId, TimestampTz timestamp)
+{
+	TupleDesc tupleDescriptor = NULL;
+	HeapTuple heapTuple = NULL;
+	Datum distributedTransactionIdDatum = 0;
+	int attributeCount = 4;
+	Datum values[attributeCount];
+	bool isNulls[attributeCount];
+
+	/* form new shard tuple */
+	memset(values, 0, sizeof(values));
+	memset(isNulls, false, sizeof(isNulls));
+
+	values[0] = ObjectIdGetDatum(databaseId);
+	values[1] = UInt64GetDatum(initiatorNodeIdentifier);
+	values[2] = UInt64GetDatum(transactionId);
+	values[3] = TimestampTzGetDatum(timestamp);
+
+	tupleDescriptor = GenerateDistributedTransactionTupleDesc();
+
+	heapTuple = heap_form_tuple(tupleDescriptor, values, isNulls);
+	distributedTransactionIdDatum = HeapTupleGetDatum(heapTuple);
+
+	return distributedTransactionIdDatum;
+}
+
+
+/*
+ * GenerateDistributedTransactionTupleDesc returns a tuple descriptor
+ * for distributed transaction id in the form of (databaseId,
+ * initiatorNodeIdentifier, transactionId, timestamp).
+ */
+static TupleDesc
+GenerateDistributedTransactionTupleDesc(void)
+{
+	TupleDesc distributedTransactionTupleDesc = NULL;
+	int attributeCount = 4;
+	bool hasOids = false;
+
+	/* generate a custom tuple descriptor for distributed transaction id*/
+	distributedTransactionTupleDesc = CreateTemplateTupleDesc(attributeCount, hasOids);
+
+	TupleDescInitEntry(distributedTransactionTupleDesc, (AttrNumber) 1, "databaseId",
+					   OIDOID, -1, 0);
+	TupleDescInitEntry(distributedTransactionTupleDesc, (AttrNumber) 2,
+					   "initiatorNodeIdentifier", INT8OID, -1, 0);
+	TupleDescInitEntry(distributedTransactionTupleDesc, (AttrNumber) 3, "transactionId",
+					   INT8OID, -1, 0);
+	TupleDescInitEntry(distributedTransactionTupleDesc, (AttrNumber) 4, "timestamp",
+					   TIMESTAMPTZOID, -1, 0);
+
+	/* let Postgres learn about this tuple descriptor */
+	BlessTupleDesc(distributedTransactionTupleDesc);
+
+	return distributedTransactionTupleDesc;
```

Comment:
We shouldn't need this - if you instead define the SRF the way I did in my prototype for the dump* functions, then you get a system provided one.